tests/test_utils_request.py

```diff
@@ -35,62 +70,51 @@ def test_request_fingerprint(self):
-                         request_fingerprint(r3, include_headers=['SESSIONID', 'Accept-Language']))
+        self.assertEqual(self.function(r3, include_headers=['accept-language', 'sessionid']),
+                         self.function(r3, include_headers=['SESSIONID', 'Accept-Language']))
 
         r1 = Request("http://www.example.com/test.html")
         r2 = Request("http://www.example.com/test.html#fragment")
-        self.assertEqual(request_fingerprint(r1), request_fingerprint(r2))
-        self.assertEqual(request_fingerprint(r1), request_fingerprint(r1, keep_fragments=True))
-        self.assertNotEqual(request_fingerprint(r2), request_fingerprint(r2, keep_fragments=True))
-        self.assertNotEqual(request_fingerprint(r1), request_fingerprint(r2, keep_fragments=True))
+        self.assertEqual(self.function(r1), self.function(r2))
+        self.assertEqual(self.function(r1), self.function(r1, keep_fragments=True))
+        self.assertNotEqual(self.function(r2), self.function(r2, keep_fragments=True))
+        self.assertNotEqual(self.function(r1), self.function(r2, keep_fragments=True))
 
         r1 = Request("http://www.example.com")
         r2 = Request("http://www.example.com", method='POST')
         r3 = Request("http://www.example.com", method='POST', body=b'request body')
 
-        self.assertNotEqual(request_fingerprint(r1), request_fingerprint(r2))
-        self.assertNotEqual(request_fingerprint(r2), request_fingerprint(r3))
+        self.assertNotEqual(self.function(r1), self.function(r2))
+        self.assertNotEqual(self.function(r2), self.function(r3))
 
         # cached fingerprint must be cleared on request copy
         r1 = Request("http://www.example.com")
-        fp1 = request_fingerprint(r1)
+        fp1 = self.function(r1)
         r2 = r1.replace(url="http://www.example.com/other")
-        fp2 = request_fingerprint(r2)
+        fp2 = self.function(r2)
         self.assertNotEqual(fp1, fp2)
 
-    def test_request_authenticate(self):
-        r = Request("http://www.example.com")
-        request_authenticate(r, 'someuser', 'somepass')
-        self.assertEqual(r.headers['Authorization'], b'Basic c29tZXVzZXI6c29tZXBhc3M=')
-
-    def test_request_httprepr(self):
-        r1 = Request("http://www.example.com")
-        self.assertEqual(request_httprepr(r1), b'GET / HTTP/1.1\r\nHost: www.example.com\r\n\r\n')
-
-        r1 = Request("http://www.example.com/some/page.html?arg=1")
-        self.assertEqual(request_httprepr(r1), b'GET /some/page.html?arg=1 HTTP/1.1\r\nHost: www.example.com\r\n\r\n')
 
-        r1 = Request("http://www.example.com", method='POST',
-                     headers={"Content-type": b"text/html"}, body=b"Some body")
-        self.assertEqual(
-            request_httprepr(r1),
-            b'POST / HTTP/1.1\r\nHost: www.example.com\r\nContent-Type: text/html\r\n\r\nSome body'
-        )
+class RequestFingerprintTest(FingerprintTest):
```

Comment:
Would it be possibile to test that the fingerprint returned by the new function is exactly the same as a fingerprint which was returned before? Non-matching fingerprints may affect media uploads, cache, etc.; that'd be a backwards incompatible change.

Reply:
The output is the same at the moment, however if we decide to fix the fingerprint generation in the new function it will not be. Assuming we will fix the fingerprint generation in the new function, what about extending the deprecation message to indicate that the new function not only returns bytes, but also a different fingerprint?